app/facades/menu/show_facade.rb

```diff
@@ -8,10 +8,18 @@ def new_order
       @order = @order.decorate
     end
 
+    def placed_order?(current_user)
+      current_user.orders.find_by(menu_id: menu.id)
+    end
+
     def orders
       @orders ||= menu.orders.decorate
     end
 
+    def show_price?(current_user)
+      current_user.admin? && menu.orders.any?
+    end
+
     def total_cost
       ordered_meals.map(&method(:current_price)).reduce(:+)
```

Comment:
```
def ordered_meals
  Meal.joins(:order).where(orders: { menu: menu })
end

def total_cost
  menu.menu_items.where(meal_id: ordered_meals.ids).sum(:current_price) # or sum(&:current_price) if that's not a field in database, but custom method
end
```
You've just produced 2 n+1 queries:
1) haven't joined meals in `ordered_meals`;
2) itterative usage of `find_by`.

[source](https://github.com/flyerhzm/bullet)

Reply:
Nice catch

Reply:
Meal.joins(:order).where(orders: { menu: menu }) seems mucho bellissimo, but I still have to count total cost from other way around - if you count it from menu, it won't count meals duplicated through orders